src/transformers/modeling_bart.py

```diff
@@ -478,6 +506,9 @@ def forward(
 
             if self.output_past:
                 next_decoder_cache.append(layer_past.copy())
+
+            if self.layer_norm and (idx == len(self.layers) - 1):  # last layer of mbart
+                x = self.layer_norm(x)
```

Comment:
I think there is something to fix on the following line (missed it on the BART PR).

The `all_hidden_states` list outputted by our models comprises `n_layer+1` elements with the first being the hidden-state after the word embedding (will be 2 times this for encoder-decoder).

I think the output of the word embeddings is missing here.

We are supposed to have a test on this so you should check the tests is activated for the encoder-decoder models (and activate/tweak it to work it if it's not).

Reply:
Good catch! I fixed `all_hidden_states` here + made an issue to fix the test, which I will address in a separate PR.